drake_ros_systems/include/drake_ros_systems/drake_ros.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef DRAKE_ROS_SYSTEMS__DRAKE_ROS_HPP_
```

Comment:
Should this be called `ros2` to clearly delineate?

(We may want to incorporate `ros1` support based on academic collaboration, after we prove out ROS2 support in Anzu.)

Reply:
I would recommend keeping the name agnostic to ROS 1 and ROS 2. For Drake users the name won't make much of a difference. That will let them migrate from one to the other without changing the includes to `drake_ros_system`, or really much of how they interact with the ROS messages themselves. I think they'd just have to change the C++ ROS message `#includes` to include `/msg/` and the `QoS` options to the system constructors.

For the ROS side, there are a few examples of suffixing the version with the name has causing some maintenance headaches in the ROS ecosystem (`ros/geometry` to `ros/geometry2`, the duplicates on the `ros2` org like `ros/kdl_parser` -> [`ros2/kdl_parser`](https://github.com/ros2/kdl_parser)).

An alternative is supporting ROS 1 and ROS 2 on different branches in the same repo (Now done on repos like [`kdl_parser`](https://github.com/ros/kdl_parser)), or supporting both ROS 1 and ROS 2 at the same time (done by some repos like [PlotJuggler](https://github.com/facontidavide/PlotJuggler)).

Reply:
Hm... The main concern is that ros1 and ros2 packages may then become unusable in the same binary.

Isn't it possible to consume them both in the same binary? (can ament consume catkin, and can ROS 1 libs avoid stepping on the feet of ROS 2 libs for symbols?)

Reply:
Ah, sorry, just read your part about the suffixing. Yeah, I think the forking of rviz2 is also part of that (though it was refactored).

OK, I'm down for trying out the forking approach first, and then pivoting later if it actually becomes a problem.
Thanks!